backend/routes/ridesharingRoutes.js

```diff
@@ -583,4 +443,201 @@ router.patch('/modify/coordinates/datetime', async(req, res) => {
+    where: { 
+      attendanceId: userToRecommendAttendance.id
+    }
+  })
+  // -----------------------------------------------
+  // Find the event data
+  event = await prisma.event.findMany({
+    where: { 
+      id: parseInt(eventId)
+    }
+  })
+  // ------------------------------------------------
+  // Find the "attendance" of the users that are driving to the event
+  try {
+      attendance = await prisma.attendance.findMany({
+        where: { 
+          eventId: parseInt(eventId),
+          ridesharingDriver: {isNot: null}
+        }
+      })
+    } catch (error) {
+      console.log(error);
+  
+      return res.status(500).json({
+        success: false,
+        message: error.message,
+      });
+  }
+  // ------------------------------------------------
+  // Store the attendance ids in an array
+  attendance.map((a) => {
+    atendanceId.push(a.id)
+  })
+  // ------------------------------------------------
+  // Search the rideshares
+  try {
+    rideshares = await prisma.ridesharing.findMany({
+        where: {
+            attendanceId: { in: atendanceId }
+        }
+    })
+  } catch (error) {
+    console.log(error);
+
+    return res.status(500).json({
+      success: false,
+      message: error.message,
+    });
+  }
+  // -----------------------------------------------
+  // Part 1 ***********************************************************
+
+  // Part 2 ***********************************************************
+  let rideshareScore = []
+
+  // We store the distance between rideshare and user
+  let distance = 0
+  let distanceEventUser = ((event.lat - userToRecommend.lat)**2 + (event.long - userToRecommend.long)**2)**.5
+  rideshares.map((rideshare) => {
+    distance = ((rideshare.departingLat - userToRecommend.lat)**2 + (rideshare.departingLong - userToRecommend.long)**2)**.5
+
+    // We penalize diferences greater the distance to the event
+    if (distance > distanceEventUser){
+      distance = 1000000
+    }
+    distanceRideshareToUser.push(distance)
+  })
+  // ----------------------------------------------
+  // We calculate how different are the date that a user is departing and the date of the rideshare
+  let difference = 0
+  rideshares.map((rideshare) => {
+    // Transform difference from ms to days
+    difference = (((rideshare.arrivalTime - userToRecommendPreferences.preferedArrivalTime)**2)**.5) / 86400000
+
+    if (difference > 1){
+      // We penalize diferences greater than a day
+      difference = 1000000
+    }
+    differenceBetweenDates.push(difference)
+  })
+  // ----------------------------------------------
+
+  // We calculate the rideshareScore 
+  let score = 0
+
+  distanceRideshareToUser.map((distance, index) => {
+    score = 1/distance + 1/differenceBetweenDates[index]
+    // If we penalized the score, we assign it a negative score
+    if (1/distance < 1/10000 || 1/differenceBetweenDates[index] < 1/10000) score = -1
```

Comment:
Is this a correct way to tell that you have penalized the score? Why not just making the penalization a flag? Also your penalization score is 1000000, and here you are checking 1/1000, there is inconsistency between.

Reply:
Yes, a flag would be an easier implementation.
Another way of solving this would me to set the penalization score to **Infinity**. But I don't know if this way is efficient.
Thank you